BlueprintUI/Sources/BlueprintView/BlueprintView.swift

```diff
@@ -370,17 +370,21 @@ public final class BlueprintView: UIView {
             )
         )
 
+        hasUpdatedViewHierarchy = true
+        isInsideUpdate = false
+
+        /// We intentionally deliver our lifecycle callbacks (eg, `onAppear`,
+        /// `onDisappear`, etc, _after_ we've marked our view as updated.
+        /// This is inc ase the `onAppear` callback triggers a re-render,
+        /// we don't hit our recurisve update precondition.
```

Comment:
I dunno, this makes me nervous. If we allow this then you can get into weird scenarios where there's a render pass with callbacks, and it forces another layout which also has callbacks, and now they're out of order — the recursive render gets to complete its callbacks before the first one. In the worst case you could kick off a loop that never allows the render to complete.

Fundamentally I'm not sure there's a strong justification for anyone needing to force an explicit re-render during `onAppear`. Why did the element need to change as a result of appearing, instead of just using that element to begin with? Why did it need to change immediately and synchronously? Seems like there are some holes in this story.